src/mac/mod.rs

```diff
@@ -857,4 +850,97 @@ where
             Err(crate::Error::Mac(Error::NetworkNotJoined))
         }
     }
+
+    fn get_send_channels<D: Device>(
+        &self,
+        device: &mut D,
+        frame: Frame,
+    ) -> [Option<<C as ChannelPlan<R>>::Channel>; NUM_OF_CHANNEL_BLOCKS] {
+        let mut channel_block_randoms = [0x00u32; NUM_OF_CHANNEL_BLOCKS];
+        for channel_block_random in channel_block_randoms.iter_mut().take(NUM_OF_CHANNEL_BLOCKS) {
+            *channel_block_random = device.rng().next_u32().unwrap_or(1);
```

Comment:
I  think this should not default to 0 if the rng fails. Better to bubble the error out to the user as something in the rng has failed

Reply:
Added the rng check and a unit test to support the change to validate_rx1_data_rate_offset().